test/kube_testing/k8s.go

```diff
@@ -268,21 +268,45 @@ func NewK8s() (*K8s, error) {
 	return &client, nil
 }
 
+// Immediate deletion does not wait for confirmation that the running resource has been terminated.
+// The resource may continue to run on the cluster indefinitely
+func (o *K8s) deletePodForce(pod *v1.Pod) error {
+	graceTimeout := int64(0)
+	delOpt := &metaV1.DeleteOptions{
+		GracePeriodSeconds: &graceTimeout,
+	}
+	err := o.clientset.CoreV1().Pods(pod.Namespace).Delete(pod.Name, delOpt)
+	if err != nil {
+		return err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), podDeleteTimeout)
+	defer cancel()
+	err = blockUntilPodWorking(o.clientset, ctx, pod)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Delete POD with completion check
+// Make force delete on timeout
 func (o *K8s) deletePods(pods ...*v1.Pod) error {
 	for _, pod := range pods {
```

Comment:
Can we try spawning `goroutines` for each of the pod to be deleted? Then, of course, wait for all fo them to finish. I hope this can speed up the termination a bit.

Reply:
Done. Now we are deleting all pods in parallel